markup_tag_evaluation/awesome_align_tokenize.py

```diff
@@ -1,23 +1,34 @@
 #!/usr/bin/env python3
 import argparse
 from io import TextIOWrapper
-from awesome_align.tokenization_bert import BertTokenizer
-from awesome_align.tokenization_utils import PreTrainedTokenizer
+from awesome_align.tokenization_bert import BertTokenizer  # type: ignore
+from awesome_align.tokenization_utils import PreTrainedTokenizer  # type: ignore
+
+UNK_TOKEN = "[UNK]"
 
 
 def write_lines(f: TextIOWrapper, texts: list[str]):
-    for l in texts:
-        f.write(l)
+    for line in texts:
+        f.write(line)
         f.write("\n")
 
 
+def get_unk_tokens(sentence: str, tokenizer: BertTokenizer) -> list[str]:
+    cleaned_sentence_ar = tokenizer.basic_tokenizer.tokenize(
+            sentence, never_split=tokenizer.all_special_tokens)
+    cleaned_sentence = "".join(cleaned_sentence_ar)
+    unk_tokens = [x for x in cleaned_sentence if x not in tokenizer.vocab]
+    return unk_tokens
```

Comment:
In awesome align the basic_tokenizer is first called here and then the individual words are tokenized: https://github.com/neulab/awesome-align/blob/5f150d45bbe51e167daf0a84abebaeb07c3323d1/awesome_align/tokenization_bert.py#L212C62-L212C97